test/ShallowWrapper-spec.js

```diff
@@ -158,6 +158,19 @@ describe('shallow', () => {
       expect(wrapper.contains(passes2)).to.equal(true);
     });
 
+    it('shouldn\'t confuse non React elements and string literals in DOM', () => {
+      const functionalComponent = () => <div></div>;
+
+      const wrapper = shallow(
+        <div>
+          <functionalComponent />
+          { 'Text node' }
+        </div>
+      );
+
+      expect(wrapper.contains(functionalComponent)).to.equal(false);
```

Comment:
Yes, that's what I'm talking about :+1: 

I would like to fix it, but:

- I don't know why it was done like that - I think somebody had some logic in mind with that, so first we need to establish that - why only ReactElements or an array of them is allowed as argument and it differs in logic so much from `find`?

- I might be a little too greenhorn in enzyme yet to make such a big change in a codebase and its logic